modules/packages/LinearAlgebra.chpl

```diff
@@ -806,14 +806,16 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
 }
 
 /*
-  Return the matrix ``A`` to the ``bth`` power, where ``b`` is a positive
-  integral type.
+  Return the matrix ``A`` to the ``bth`` power, where ``b`` is am integer.
 
   .. note::
 
     ``matPow`` will utilize the :mod:`BLAS` module for improved performance, if
     available. Compile with ``--set blasImpl=none`` to opt out of the
     :mod:`BLAS` implementation.
+
+    If ``b`` is negative, ``matPow`` will depend on :mod:`LAPACK` module, and will
+    generate a compiler error if ``lapackImpl`` is ``none``.
```

Comment:
After your change, this is not the case anymore.

Reply:
Is that because it won't be a compiler error?

Reply:
Yes. After this last change, you stop calling `inv` but call `_expBySquaring`, instead. And that'll halt instead of failing at compile time.

And by looking at it again, I would think that your previous implementation would have failed on a system with no LAPACK. Because adding that `usingLapack` you just told the compiler that if there is no LAPACK in the system, then don't even compile this block. Before that change you were attempting to compile `inv` always, which should fail `usingLapack` is false.

Reply:
Is my updated description accurate?